pkgs/by-name/hy/hyprwall/package.nix

```diff
@@ -0,0 +1,85 @@
+{
+  lib,
+  fetchurl,
+  fetchFromGitHub,
+  rustPlatform,
+  makeDesktopItem,
+  copyDesktopItems,
+
+  pkg-config,
+  glib,
+  gdk-pixbuf,
+  pango,
+  graphene,
+  gtk4,
+  gtk3,
+  git,
+}:
+
+rustPlatform.buildRustPackage rec {
+  pname = "hyprwall";
+  version = "0.1.8";
+
+  src = fetchFromGitHub {
+    owner = "hyprutils";
+    repo = "hyprwall";
+    rev = "refs/tags/v${version}";
+    hash = "sha256-dzPd+5cws3hKhdd1CKKEO7EWMS0XW0y1vqxg1XKX+gY=";
+  };
+
+  cargoHash = "sha256-gT2ysWHckcUl1yx5tciy6kSvZZ0srrs4OwI1mr/58Pc=";
+
+  desktopItems = [
+    (makeDesktopItem {
+      name = "HyprWall";
+      exec = "hyprwall";
+      icon = fetchurl {
+        url = "https://github.com/hyprutils/hyprwall/blob/main/hyprwall.png";
+        hash = "sha256-YkggQGJiN3TaBTW8aaorBzl5pJwf9fiU0ZnLCj6knhk=";
+      };
+      type = "Application";
+      terminal = false;
+      comment = "GUI for setting wallpapers with Hyprpaper, written in blazingly fast Rust!";
+      desktopName = "HyprWall";
+      categories = [
+        "Utility"
+      ];
+      keywords = [
+        "wallpaper"
+        "hyprwall"
+      ];
+    })
+  ];
```

Comment:
HyprWall already has [`hyprwall.desktop`](https://github.com/hyprutils/hyprwall/blob/main/hyprwall.desktop) in its source, just use it.

Anyway, it's better to copy the icon from the source rather than download it from **main** (by the way, the icon hash is incorrect too).

Reply:
I don't really know how to override the install phase to store the binary in the bin folder, hence the workaround

Reply:
Not the best, but a working **example** of `postInstall`:
```nix
postInstall = ''
  mkdir -p "$out/share/applications"
  mkdir -p "$out/share/icons"
  cp hyprwall.desktop "$out/share/applications"
  cp hyprwall.png "$out/share/icons"
  substituteInPlace "$out/share/applications/hyprwall.desktop" --replace 'Exec=/usr/bin/hyprwall' "Exec=$out/bin/hyprwall"
'';
```